src/main/scala/SubscriptionActor.scala

```diff
@@ -0,0 +1,84 @@
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import akka.util.Timeout
+import generic.Event
+import sangria.ast.OperationType
+import sangria.execution.{Executor, PreparedQuery}
+import sangria.marshalling.sprayJson._
+import sangria.parser.QueryParser
+import spray.json._
+
+import scala.util.{Failure, Success}
+
+object SubscriptionActor extends DefaultJsonProtocol {
+  sealed trait SubscriptionMessage
+  case class Subscribe(query: String, operation: Option[String])
+  case class SubscriptionAccepted() extends SubscriptionMessage
+  case class QueryResult(json: JsValue) extends SubscriptionMessage
+  case class Connected(outgoing: ActorRef)
+  case class PreparedQueryContext(query: PreparedQuery[Ctx, Any, JsObject])
+
+  implicit val subscribeProtocol = jsonFormat2(Subscribe)
+}
+
+class SubscriptionActor(publisher: ActorRef, ctx: Ctx)(implicit timeout: Timeout) extends Actor with ActorLogging {
+
+  import SubscriptionActor._
+
+  implicit val ec = context.system.dispatcher
+
+  val executor = Executor(schema.createSchema)
+  var subscriptions = Map.empty[String, Set[PreparedQueryContext]]
+
+  override def receive: Receive = {
+    case Connected(outgoing) =>
+      publisher ! SubscriptionEventPublisher.Join
+      context.become(connected(outgoing))
+  }
+
+  def connected(outgoing: ActorRef): Receive = {
+    case e: Subscribe =>
+      log.info(s"Got sub: $e")
+      prepareQuery(e)
+
+    case context: PreparedQueryContext =>
```

Comment:
I'm using a wrapper class. In my project, I need to pass around the variables with the subscription.